### ISSUE TITLE
the gradient accumulate not correctly used?

### ISSUE THREAD

**Issue by rassaire** [mobaidoctor/med-ddpm#53]
Hello,
I am wondering whether ```accumulate_gradient``` is being used correctly. In the trainer, I found the following code: it seems that the batch loss is averaged and then also divided again by the gradient accumulation factor — all inside the loop. I was expecting the accumulated loss (```average_loss```) to be used for the backward pass outside the loop.

```python

while self.step < self.train_num_steps:
            accumulated_loss = []
            for i in range(self.gradient_accumulate_every):
                if self.with_condition:
                    data = next(self.dl)
                    input_tensors = data['input'].cuda()
                    target_tensors = data['target'].cuda()
                    loss = self.model(target_tensors, condition_tensors=input_tensors)
                else:
                    data = next(self.dl).cuda()
                    loss = self.model(data)
                loss = loss.sum()/self.batch_size
                print(f'{self.step}: {loss.item()}')
                backwards(loss / self.gradient_accumulate_every, self.opt)
                accumulated_loss.append(loss.item())
# Record here
 average_loss = np.mean(accumulated_loss)
```
